### PR TITLE
Nicklathe/fix dynamic registration

### PR DESCRIPTION
**Pull request by nicklathe**
This PR address a few Dynamic Registration improvements and bug fixes.

Fixes:
* Specifies the same scopes in registration as we request with the authentication token. This was preventing course syncing from working, as we were requesting different scopes in our authentication token request than we were specifying during the registration step.

Enhancements:
* Adds padding between the email label in the input text box
* Adds icon URL
* Removes undesirable placements
* Uses the `CDO.studio_url` and `CDO.dashboard_site_host` helper to set dynamic domain based on environment

Cleanup:
* Removes unused imports left over from a previous dynamic registration PR

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/lib/policies/lti.rb [hunk 1]
```diff
@@ -90,11 +90,11 @@ module MessageType
     jwks_uri: CDO.studio_url('/oauth/jwks', CDO.default_scheme),
     token_endpoint_auth_method: "private_key_jwt",
     contacts: ["platform@code.org"],
-    scope: "https://purl.imsglobal.org/spec/lti-ags/scope/score https://purl.imsglobal.org/spec/lti-nrps/scope/contextmembership.readonly",
+    scope: ALL_SCOPES.join(' '),
     "https://purl.imsglobal.org/spec/lti-tool-configuration" => {
-      domain: "studio.code.org",
-      description: "Code.org",
-      target_link_uri: "https://studio.code.org/lti/v1/sync_course",
+      domain: CDO.studio_url('', CDO.default_scheme),
```
Comment (stephenliang) [inline dashboard/lib/policies/lti.rb]: Per the [LTI spec](https://www.imsglobal.org/spec/lti-dr/v1p0#lti-configuration-0) it looks like domain shouldn't include the protocol which I believe this change will add. 

> The primary domain covered by this tool; protocol must not be included. For example mytool.example.org

Comment (nicklathe) [reply dashboard/lib/policies/lti.rb]: Hmmm yeah you're right. I'll look into if we have a helper method that leaves the protocol off.

Comment (nicklathe) [reply dashboard/lib/policies/lti.rb:95]: Looks like I can just leave the second argument blank, and it defaults to an empty string.

Comment (nicklathe) [reply dashboard/lib/policies/lti.rb]: Looks like I should use [CDO.dashboard_site_host](https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo.rb#L114)

Comment (nicklathe) [reply dashboard/lib/policies/lti.rb]: Tested on multiple environments:

Prod
```
[production] dashboard > CDO.dashboard_site_host
=> "studio.code.org"
```

Test
```
[test] dashboard > CDO.dashboard_site_host
=> "test-studio.code.org"
```

Local development
```
[development] dashboard > CDO.dashboard_site_host
=> "localhost-studio.code.org:3000"
```

File: dashboard/lib/policies/lti.rb [hunk 2]
```diff
@@ -90,11 +90,11 @@ module MessageType
...
     contacts: ["platform@code.org"],
-    scope: "https://purl.imsglobal.org/spec/lti-ags/scope/score https://purl.imsglobal.org/spec/lti-nrps/scope/contextmembership.readonly",
+    scope: ALL_SCOPES.join(' '),
     "https://purl.imsglobal.org/spec/lti-tool-configuration" => {
-      domain: "studio.code.org",
-      description: "Code.org",
-      target_link_uri: "https://studio.code.org/lti/v1/sync_course",
+      domain: CDO.dashboard_site_host,
+      description: "Code.org LTI Integration",
+      target_link_uri: CDO.studio_url('/lti/v1/sync_course', CDO.default_scheme),
```
Comment (carl-codeorg) [inline dashboard/lib/policies/lti.rb]: Does `lti_v1_sync_course_path` work for the path segment here? Could also add a new const DEFAULT_TARGET_LINK_URI and set it to `lti_v1_sync_course_path`

**Review by carl-codeorg — APPROVED** [review]
One tiny non-blocking comment, otherwise LGTM

File: dashboard/lib/policies/lti.rb (hunk 2 above)
Comment (nicklathe) [reply dashboard/lib/policies/lti.rb]: `lti_v1_sync_course_path` doesn't seem to work here. I like the idea of adding a new const in this file and referencing it.

**Pull request merged**